### PR TITLE
add userstory10 / overview-page

### PR DESCRIPTION
**Pull request by rabe200**
created overview pages

### CODE REVIEW

File: pages/cards/overview/big-arcana/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  const [hasMounted, setHasMounted] = useState(false);
+
+  const cards = useStore((state) => state.allCards);
+
+  useEffect(() => {
+    setHasMounted(true);
+  }, []);
+  if (!hasMounted) {
+    return null;
+  }
```
Comment (PReichetanz) [inline pages/cards/overview/big-arcana/index.js:16]: I don't understand this part, can you please explain to me why you need this?

File: pages/cards/overview/big-arcana/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  return (
+    <>
+      <ul>
+        <h2>big arcana</h2>
```
Comment (PReichetanz) [inline pages/cards/overview/big-arcana/index.js:20]: The direct child of an `ul` element should be a `li`, so please move the `h2` above the list.

File: pages/cards/overview/big-arcana/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          .map((card) => (
+            <li key={card.id}>
+              {" "}
+              <Link href={{ pathname: `/cards/${card.id}` }}>{card.name}</Link>
+            </li>
+          ))}
+      </ul>
+      <BackButton />
+      <Link href="/">
+        <button type="button">menu</button>
```
Comment (PReichetanz) [inline pages/cards/overview/big-arcana/index.js:32]: Please remove the `button` here as well.

File: pages/cards/overview/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import BackButton from "../../../components/Backbutton/backbutton";
+export default function Description() {
+  const [hasMounted, setHasMounted] = useState(false);
+
+  useEffect(() => {
+    setHasMounted(true);
+  }, []);
+  if (!hasMounted) {
+    return null;
+  }
```
Comment (PReichetanz) [inline pages/cards/overview/index.js:13]: I don't understand the usage here either.

File: pages/cards/overview/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  return (
+    <>
+      <ul>
+        <li>
+          <Link href={{ pathname: "/cards/overview/big-arcana" }}>
+            big arcana
```
Comment (PReichetanz) [inline pages/cards/overview/index.js:19]: This is a navigation, isn't it? If so, use a `nav` tag to wrap it.

File: pages/cards/overview/index.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+          <li>
+            <Link href={{ pathname: "/cards/overview/minor-arcana/swords" }}>
+              swords
+            </Link>
+          </li>
+        </ul>
+      </ul>
+      <BackButton />
+      <Link href="/">
+        <button type="button">menu</button>
```
Comment (PReichetanz) [inline pages/cards/overview/index.js:52]: Please remove the `button` here as well.

File: pages/cards/overview/minor-arcana/cups/index.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  useEffect(() => {
+    setHasMounted(true);
+  }, []);
+  if (!hasMounted) {
+    return null;
+  }
+  return (
+    <>
+      <ul>
+        <h2>cups</h2>
```
Comment (PReichetanz) [inline pages/cards/overview/minor-arcana/cups/index.js:20]: Please move the `h2` above the `ul`.

File: pages/cards/overview/minor-arcana/index.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  if (!hasMounted) {
+    return null;
+  }
+  return (
+    <>
+      <ul>
+        <h1>minor arcana</h1>
+        <li>
+          {" "}
+          <Link href={{ pathname: "/cards/overview/minor-arcana/wands" }}>
```
Comment (PReichetanz) [inline pages/cards/overview/minor-arcana/index.js:20]: I think the pathnames should be generated dynamically as well as the corresponding pages to which they link.

File: pages/cards/overview/minor-arcana/pentacles/index.js
Comment (PReichetanz) [inline pages/cards/overview/minor-arcana/pentacles/index.js:1]: Please consider to create a dynamic route for all folders inside of `pages/minor-arcana`(I have no idea what a good name for this dynamic route could be due to my lack of knowledge in this area).

File: pages/cards/overview/minor-arcana/swords/index.js
Comment (PReichetanz) [inline pages/cards/overview/minor-arcana/swords/index.js:1]: This page should be generated dynamically as well.

File: pages/cards/overview/minor-arcana/wands/index.js
Comment (PReichetanz) [inline pages/cards/overview/minor-arcana/wands/index.js:1]: This page should be generated dynamically as well.

File: styles.js [hunk 9]
```diff
@@ -8,12 +8,16 @@ export default createGlobalStyle`
   }
 
   body {
-    margin: 0;
+    /* margin: 0;
```
Comment (PReichetanz) [inline styles.js:11]: If you don't need it, remove it.

**Review by PReichetanz — CHANGES_REQUESTED** [review]
This is a good starting point! Please consider to use dynamic routes to follow the DRY rule and have a look at my comments on semantics.

File: pages/cards/overview/minor-arcana/pentacles/index.js
Comment (rabe200) [reply pages/cards/overview/minor-arcana/pentacles/index.js:1]: yeah i was thinking about the dynamic routing directly after  opening the pr - gonna work on it now

File: pages/cards/overview/big-arcana/index.js (hunk 1 above)
Comment (rabe200) [reply pages/cards/overview/big-arcana/index.js:16]: copy paste from another page where i was working with rng content - not needed here, as client and ssr aren't differing. thanks

File: pages/cards/overview/minor-arcana/index.js (hunk 8 above)
Comment (rabe200) [reply pages/cards/overview/minor-arcana/index.js]: @PReichetanz done

File: pages/cards/overview/minor-arcana/swords/index.js
Comment (rabe200) [reply pages/cards/overview/minor-arcana/swords/index.js:1]: @PReichetanz done

File: pages/cards/overview/minor-arcana/wands/index.js
Comment (rabe200) [reply pages/cards/overview/minor-arcana/wands/index.js:1]: @PReichetanz done

**Review by PReichetanz — APPROVED** [review]

**Pull request merged**